scriptrenderer.go

```diff
@@ -33,18 +36,26 @@ func (r *scriptRenderer) renderNoop(w util.BufWriter, source []byte, node ast.No
 
 func (r *scriptRenderer) renderCodeBlock(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
 	if entering {
-		filepath := ""
+		path := ""
 		for i := 0; i < node.Lines().Len(); i++ {
 			line := node.Lines().At(i)
 			value := line.Value(source)
 			if i == 0 {
-				if !bytes.HasPrefix(value, []byte("### FILE: ")) {
-					continue
-				} else {
-					filepath = strings.TrimSpace(string(value[10:]))
+				filePragmaRE.Longest()
+				if match := filePragmaRE.FindSubmatch(value); match != nil {
+					p := filepath.ToSlash(string(match[1]))
+					switch {
+					case filepath.IsAbs(p):
+						fmt.Printf("Warning: absolute paths are not allowed, ignoring path: %s\n", p)
+					case filepath.Clean("/"+p) != "/"+p:
+						fmt.Printf("Warning: using .. in paths is not allowed, ignoring path: %s\n", p)
+					default:
+						// accept this path
+						path = p
+					}
```

Comment:
I think in the error cases the code should actually *exit* the function by returning. Otherwise the output will be cluttered with an empty key containing all ignored codeblocks which is pretty useless. This was already an oversight in the original implementation, but since we're already improving here...

Reply:
Agreed, I enhanced the PR, it's ready to review :-)